test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -656,4 +685,22 @@ def with_custom_login_url(url)
   ensure
     ShopifyApp.configure { |config| config.login_url = original_url }
   end
+
+  def with_shopify_security_base_current_tenant
+    shopify_security_base_module = Module.new
+    ShopifyApp::LoginProtection.const_set("ShopifySecurityBase", shopify_security_base_module)
```

Comment:
I couldn't find a nicer way to stub the existence of a class than using `const_set` then `remove_const` (which for some reason is private even though `const_set` isn't). Open to other ideas if anyone has them!